src/FixedDataTable.js

```diff
@@ -773,14 +773,17 @@ var FixedDataTable = React.createClass({
     var columnBefore = this.state.columnReorderingData.columnBefore;
     var columnAfter = this.state.columnReorderingData.columnAfter;
     var reorderColumn = this.state.columnReorderingData.columnKey;
+    var cancelReorder = this.state.columnReorderingData.cancelReorder;
 
     this.setState({
       isColumnReordering: false,
       columnReorderingData: {}
     });
-    this.props.onColumnReorderEndCallback({
-      columnBefore, columnAfter, reorderColumn
-    });
+    if (!cancelReorder) {
+      this.props.onColumnReorderEndCallback({
+        columnBefore, columnAfter, reorderColumn
+      });
+    }
 
     var onHorizontalScroll = this.props.onHorizontalScroll;
```

Comment:
Good call. I only made sure that we want the stuff above the cancelReorder check, I didn't look below it to see if we want to skip anything.

I'll change the contents of the if block to return early and put the Callback after the if block.